src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from 'react';
+import { Device, AudioTransformDevice, VoiceFocusTransformDevice } from 'amazon-chime-sdk-js';
+
+import { ControlBarButton } from '../../ui/ControlBar/ControlBarButton';
+import { Microphone } from '../../ui/icons';
+import { useMeetingManager } from '../../../providers/MeetingProvider';
+import { useAudioInputs } from '../../../providers/DevicesProvider';
+import { useVoiceFocus } from '../../../providers/VoiceFocusProvider';
+import { useToggleLocalMute } from '../../../hooks/sdk/useToggleLocalMute';
+import { DeviceConfig } from '../../../types';
+import { isOptionActive, audioInputSelectionToDevice } from '../../../utils/device-utils';
+import { PopOverItemProps } from '../../ui/PopOver/PopOverItem';
+import { PopOverSeparator } from '../../..';
+import { Spinner } from '../../ui/icons';
+
+interface Props {
+  /** The label that will be shown when microphone is muted , it defaults to `Mute`. */
+  muteLabel?: string;
+  /** The label that will be shown when microphone is unmuted, it defaults to `Unmute`. */
+  unmuteLabel?: string;
+  /** Title attribute for the icon when muted, it defaults to `Muted microphone` */
+  mutedIconTitle?: string;
+  /** Title attribute for the icon when unmuted, it defaults to `Microphone` */
+  unmutedIconTitle?: string;
+}
+
+const AudioInputVFControl: React.FC<Props> = ({
+  muteLabel = 'Mute',
+  unmuteLabel = 'Unmute',
+  mutedIconTitle,
+  unmutedIconTitle
+}) => {
+  const meetingManager = useMeetingManager();
+  const [ voiceFocusOn, setVoiceFocusOn ] = useState(false);
+  const [ isLoading, setIsloading ] = useState(false);
+  const [ enableVoiceFocus, setEnableVoiceFocus] = useState(false);
+  const { muted, toggleMute } = useToggleLocalMute();
+  const [ device, setDevice ] = useState<Device | AudioTransformDevice | null>(meetingManager.selectedAudioInputTransformDevice);
+  const { supportsVoiceFocus, transformToVoiceFocus } = useVoiceFocus();
+  const audioInputConfig: DeviceConfig = {
+    additionalDevices: true,
+  };
+  const { devices, selectedDevice } = useAudioInputs(audioInputConfig);
+
+  useEffect(() => {
+    if (device instanceof VoiceFocusTransformDevice) {
+        setEnableVoiceFocus(true);
+    } else {
+        setEnableVoiceFocus(false);
+    }
+  }, [device]);
+
+  useEffect(() => {
+    const callback = (updatedAudioInputDevice: Device | AudioTransformDevice | null): void => {
+        setDevice(updatedAudioInputDevice);
+    };
+    meetingManager.subscribeToSelectedAudioInputTransformDevice(callback);
+
+    return (): void => {
+      meetingManager.unsubscribeFromSelectedAudioInputTransformDevice(callback);
+    };
```

Comment:
Can you just use `setDevice` instead of `callback` in these lines?

Reply:
Sure, updated. We use this pattern through out, like in [AudioInputProvider](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/master/src/providers/DevicesProvider/AudioInputProvider.tsx#L36), [AudioVideoProvider](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/master/src/providers/AudioVideoProvider/index.tsx#L18). Do I need to update all of them?

Reply:
Update other `callback`s throughout